test/conic_program.jl

```diff
@@ -38,80 +39,103 @@ function _test_simple_socp(eq_vec::Bool)
+                ),
+                dx,
+                atol = ATOL,
+                rtol = RTOL,
+            ),
         )
     else
-        MOI.set(model, DiffOpt.ForwardConstraintFunction(), ceq, 1.0 * x)
+        @test JuMP.coefficient(
+            MOI.get(model, DiffOpt.ReverseConstraintFunction(), ceq),
+            x,
+        ) ≈ dx atol = ATOL rtol = RTOL
     end
-    MOI.set(
-        model,
-        DiffOpt.ForwardConstraintFunction(),
-        cnon,
-        MOI.Utilities.vectorize([1.0 * y]),
-    )
+
+    DiffOpt.empty_input_sensitivities!(model)
+
+    MOI.set(model, DiffOpt.ForwardConstraintFunction(), cnon, 1.0 * y)
+
+    DiffOpt.forward_differentiate!(model)
+
+    dy = -0.707083
+    @test MOI.get(model, DiffOpt.ForwardVariablePrimal(), y) ≈ dy atol = ATOL rtol =
+        RTOL
+
+    MOI.set(model, DiffOpt.ReverseVariablePrimal(), y, 1.0)
+
+    DiffOpt.reverse_differentiate!(model)
+
+    @test JuMP.coefficient(
+        MOI.get(model, DiffOpt.ReverseConstraintFunction(), cnon),
+        y,
+    ) ≈ dy atol = ATOL rtol = RTOL
+
+    DiffOpt.empty_input_sensitivities!(model)
+
     MOI.set(
         model,
         DiffOpt.ForwardConstraintFunction(),
         csoc,
-        MOI.Utilities.operate(vcat, Float64, 1.0 * t, 0.0, 0.0),
+        MOI.Utilities.operate(vcat, Float64, 1.0 * t.index, 0.0, 0.0),
     )
+
     DiffOpt.forward_differentiate!(model)
-    # these matrices are benchmarked with the output generated by diffcp
-    # refer the python file mentioned above to get equivalent python source code
-    @test model.diff.model.x ≈ [-1 / √2; 1 / √2; 1.0] atol = ATOL rtol = RTOL
-    if eq_vec
-        @test model.diff.model.s ≈ [0.0, 0.0, 1.0, -1 / √2, 1 / √2] atol = ATOL rtol =
-            RTOL
-        @test model.diff.model.y ≈ [√2, 1.0, √2, 1.0, -1.0] atol = ATOL rtol =
-            RTOL
-    else
-        @test model.diff.model.s ≈ [0.0, 1.0, -1 / √2, 1 / √2, 0.0] atol = ATOL rtol =
-            RTOL
-        @test model.diff.model.y ≈ [1.0, √2, 1.0, -1.0, √2] atol = ATOL rtol =
-            RTOL
-    end
-    dx = [1.12132144; 1 / √2; 1 / √2]
-    for (i, vi) in enumerate([x, y, t])
-        @test dx[i] ≈ MOI.get(model, DiffOpt.ForwardVariablePrimal(), vi) atol =
-            ATOL rtol = RTOL
-    end
-    # @test dx ≈ [1.12132144; 1/√2; 1/√2] atol=ATOL rtol=RTOL
-    # @test ds ≈ [0.0; 0.0; -2.92893438e-01;  1.12132144e+00; 7.07106999e-01]  atol=ATOL rtol=RTOL
-    # @test dy ≈ [2.4142175; 5.00000557; 3.8284315; √2; -4.00000495] atol=ATOL rtol=RTOL
+
+    ds = 0.0
+    @test MOI.get(model, DiffOpt.ForwardVariablePrimal(), t) ≈ ds atol = ATOL rtol =
+        RTOL
+
+    MOI.set(model, DiffOpt.ReverseVariablePrimal(), t, 1.0)
+
+    DiffOpt.reverse_differentiate!(model)
+
+    # FIXME: this is not working - https://github.com/jump-dev/DiffOpt.jl/issues/283
+    # @test JuMP.coefficient(MOI.get(model, DiffOpt.ReverseConstraintFunction(), csoc).func.func.func, t.index) ≈ ds atol=ATOL rtol=RTOL
+
```

Comment:
No clear API: https://github.com/jump-dev/DiffOpt.jl/issues/283

Reply:
this will be left open for a future PR